### ISSUE TITLE
Allow KerasModel to accept tf.keras.optimizers objects directly

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1952]
At present, `KerasModel` expects specified optimizers to be instances of `dc.models.optimizers.Optimizer`. These are wrappers around `tf.keras.optimizers` optimizers that provide the nice ability to be able to define a `LearningRateSchedule` for your learning rate. 

At present `KerasModel` throws an error if you try to use it with a `tf.keras.optimizers` optimizer out of the box. I propose that we allow `KerasModel` to direclty accept `tf.keras.optimizers` objects. This is an easier default for beginners who can later define their own `Optimizer` objects if they'd like something more customizable.
